shuup/admin/views/select.py

```diff
@@ -193,3 +200,68 @@ def _filter_query(self, request, cls, qs, shop, search_mode=None):
 
     def get(self, request, *args, **kwargs):
         return JsonResponse({"results": self.get_data(request, *args, **kwargs)})
+
+
+class ObjectSelectorView(TemplateView):
+    """
+    Base class for responding to searches from select2 components.
+    """
+
+    def get(self, request, *args, **kwargs):
+
+        selector = request.GET.get("selector")
+        search_term = request.GET.get("q").strip()
+        search_mode = request.GET.get("searchMode")
+        sales_units = request.GET.get("salesUnits")
+        user = request.user
+
+        shop = request.GET.get("shop")
+        if shop:
+            query_shop = Shop.objects.get_for_user(request.user).filter(pk=request.GET["shop"]).first()
+            if query_shop:
+                shop = query_shop
+        else:
+            shop = Shop.objects.get_for_user(request.user).first()
```

Comment:
Force a shop defined to comply with the tests. In case there are multiple vendors, it is taking only one. Is this a problem?

Reply:
the shop here should be always `get_shop(request)`